### ISSUE TITLE
MD029 doesn't allow for start="" in ol tags

### ISSUE THREAD

**Issue by af4jm** [DavidAnson/markdownlint#340]
if I set MD029 to "ordered", I would expect any list item to get flagged if it's not 1 more than the previous number, but if I do something like...

``` markdown
# heading

## subheading 1

1. blah
2. blah
3. blah

## subheading 2

4. blah
5. blah
6. blah
```

this is "ordered", e.g. the 5. should NOT be flagged because the previous is 4... yet it is flagged and the hover text says that the setting is "1-2-3"... um, no, it's not "ordered starting with 1" it's just "ordered"... and it renders correct & valid html with start="4" on the second ol tag (I will admit this type of list only makes sense when the items in the list are naturally numbered anyway... e.g the 10 Commandments, where the subheadings would be "God" & "neighbors")

but it seems odd that markdownlint has no way of creating a start attribute without it being flagged as a violation

**Comment by af4jm** [comment]
Hmm... that's an interesting perspective... I see only 3 options...

1. first item can be any number, subsequent numbers increase by 1
2. first item can be anything, subsequent items match
3. linter suggests a change that alters the output (by claiming that a start attribute in an ol tag is an error)

in my opinion #3 smells like a bug, although the simple fact that this project has been around for a while working the way it is indicates I'm probably very much in the minority, so I'm going to turn this into a question...

if I were going to write a custom rule to work the way I want it to work, what would my filter look like to get all the lists? I tried to start with a copy of MD029, but it uses cache.js which probably isn't accessible from a custom rule (at least not the instance that's actually populated with the document tree). I also tried grabbing ol tags, but I think I also need the child li tags as well. 
Looking through the other example rules linked didn't help, presumably because what I'm trying to do is very uncommon.

Thanks!

**Comment by DavidAnson** [comment]
I'd suggest starting from - or modelling - MD029. `cache.js` is not needed - the helpers in https://www.npmjs.com/package/markdownlint-rule-helpers should help with some of the grunt work. You can see what the parse tree looks like for different syntax by looking at the "debug" tab on https://markdown-it.github.io/.

I'll caution you that your item 1 would allow "2/3/4" which is usually wrong and item 2 would allow "5/5/5" which is probably also wrong. You may want to remember previous lists and allow starting with a number other than 1 iff the previous list ended with the previous value.

Good luck!

**Comment by af4jm** [comment]
Thanks... and point taken on your "usually wrong" comments, which 100% counter my arguments above... so far I have a custom rule doing exactly what MD029 does out of the box, once I get it doing the custom thing (not today, but probably this weekend) I'll stick it in a gist & link to it here for future reference... but I have definitely been swayed that the existing behavior of MD029 is what makes sense for most cases

**Comment by nschonni** [comment]
@af4jm unless this style is all over your docs, it might be easier to just use `<--markdownlint-disable MD029-->` and `<--markdownlint-enable MD029-->` comments around specific blocks you're sure about

**Comment by DavidAnson** [comment]
@af4jm Feel free to publish your rule to npm. If you tag it (see CustomRules.md), others can find it.

**Comment by af4jm** [comment]
published, thanks!

@nschonni I didn't want to do that because the repo in question is an Azure DevOps wiki I'm using to replace my Evernote account (partly because it's the only Markdown-based wiki I could find... and being free was just a bonus.)

**Comment by DavidAnson** [comment]
Congrats! I'll leave this issue open to consider enhancing MD029 with an option to allow list continuations.

**Comment by DavidAnson** [comment]
Closing this based on it usually being an error - and because there's now a rule to help with this: https://www.npmjs.com/package/markdownlint-rule-olstart.
